jupyter_scheduler/orm.py

```diff
@@ -108,6 +109,7 @@ class JobDefinition(CommonColumns, Base):
     url = Column(String(256), default=generate_job_definitions_url)
     create_time = Column(Integer, default=get_utc_timestamp)
     active = Column(Boolean, default=True)
+    notification = Column(JsonType(1024), nullable=True)
```

Comment:
Thanks for pointing this out. It would have been best to address this during the design review, but as discussed there was not very much time to dive deep into all the aspects of the design doc.

Reply:
Added a separate table for `NotificationsConfig` (note that eager loading relevant `NotificationsConfig` object is required to prevent `DetachedInstanceError` that occurs when Job instance detached from the session tries to access notifications config). There is no native SQLAlchemy data type that covers lists of strings with necessary granularity (both number of items in the list and size of each item) so I opted for enforcing constraints on application level through [pydantic validators](https://github.com/jupyter-server/jupyter-scheduler/pull/446/files#diff-20dd20fee14818d8deb4f44eb356356178e132f8b838a28c0a4027a0e73ac073R43-R67).